File: testerum-backend/testerum-runner/testerum-runner-api/src/main/kotlin/com/testerum/runner/events/model/RunnerErrorEvent.kt
package com.testerum.runner.events.model

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.testerum.model.runner.tree.id.RunnerIdCreator
import java.time.LocalDateTime

/*
 * This event is never emitted by the runner.
 * It is only used to communicate between the backend and the frontend.
 */
data class RunnerErrorEvent @JsonCreator constructor(
        @JsonProperty("time")         override val time: LocalDateTime = LocalDateTime.now(),
        @JsonProperty("errorMessage") val errorMessage: String
): RunnerEvent {

    @get:JsonProperty("eventKey")
    override val eventKey = RunnerIdCreator.getRootId()

}

File: web/web-frontend/src/app/model/test/event/runner-error.event.ts
import {RunnerEvent} from "./runner.event";
import {RunnerEventTypeEnum} from "./enums/runner-event-type.enum";
import {Serializable} from "../../infrastructure/serializable.model";

export class RunnerErrorEvent implements RunnerEvent, Serializable<RunnerErrorEvent> {
    time: Date;
    eventKey: string;
    eventType: RunnerEventTypeEnum = RunnerEventTypeEnum.ERROR_EVENT;

    errorMessage: string;

    deserialize(input: Object): RunnerErrorEvent {
        this.eventKey = input["eventKey"];
        this.time = new Date(input["time"]);
        this.errorMessage = input["errorMessage"];
        return this;
    }

    serialize(): string {
        return undefined;
    }
}
